### PR TITLE
Hirshfeld Automation

### PR DESCRIPTION
**Pull request by OscarvanVuren**
A few scripts for automated hirshfeld extraction from FHI-aims.

### CODE REVIEW

**Comment by OscarvanVuren** [comment]
Needs ci tests for the new functionality. This is coming soon

**Review by GaryLZW — APPROVED** [review]
I left some minor comments. This looks fine to me.

File: carmm/analyse/hirshfeld.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        identifier = 'Free atom volume        :'
+    elif data == 'dipole vector':
+        identifier = 'Hirshfeld dipole vector :'
+    elif data == 'dipole moment':
+        identifier = 'Hirshfeld dipole moment :'
+    elif data == 'second':
+        identifier = 'Hirshfeld second moments:'
+    else:
+        print('Requested data not recognised')
+        sys.exit()
```
[the remark this answers is not in the document]
Comment (OscarvanVuren) [reply carmm/analyse/hirshfeld.py:35]: Thanks, good suggestion and changes made

File: carmm/analyse/hirshfeld.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            hirshfeld_line = output_line
+
+        else:
+            output_line -= 1
+
+    return hirshfeld_line
+
+
+def write_hirshfeld(text, natoms, start, outname):
+
```
Comment (AkashHiregange) [inline carmm/analyse/hirshfeld.py:72]: It would be nice if you could write a docstring for this function. I got confused about the 'start' argument.

Comment (OscarvanVuren) [reply carmm/analyse/hirshfeld.py:72]: Its an internal function that shouldn't really be seen by a user, but I agree that it is currently opaque

**Pull request merged**